### PR TITLE
feature/[PATRO23-64] - keycloak user authentication

### PR DESCRIPTION
**Pull request by pzajaczkowski**
https://tracker.intive.com/jira/browse/PATRO23-64

### CODE REVIEW

File: src/api/app/Program.cs [hunk 1]
```diff
@@ -33,39 +34,27 @@
...
+builder.Services.AddSwagger();
 
 builder.Services.AddControllers();
 
 builder.Services.AddFromAssemblies(typeof(IDomainEventHandler<>));
 builder.Services.AddFromAssemblies(typeof(IEventDispatcher<>));
 builder.Services.AddFromAssemblies(typeof(ICommandHandler<>));
 builder.Services.AddFromAssemblies(typeof(IQueryHandler<,>));
+
+builder.Services.AddKeycloakAuthentication(builder.Configuration, configureOptions =>
```
Comment (mateuszgiedzinski) [inline src/api/app/Program.cs:46]: Try to not put blocks of code between block that is for one funciton. here you separate commands and queries registration from bus registration.

File: src/api/app/Program.cs [hunk 2]
```diff
@@ -76,4 +65,13 @@
 
 app.UseExampleModule();
 
+app.UseAuthentication();
+app.UseAuthorization();
+
+if (app.Environment.IsDevelopment())
```
Comment (mateuszgiedzinski) [inline src/api/app/Program.cs:71]: Please remove this IsDevelopemet check.. It's nothing wrong to have swagger on production for now.

File: src/modules/example/api/Controllers/ExampleController.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 using Intive.Patronage2023.Shared.Abstractions;
 using Intive.Patronage2023.Shared.Abstractions.Commands;
 using Intive.Patronage2023.Shared.Abstractions.Queries;
-
+using Microsoft.AspNetCore.Authorization;
```
Comment (mateuszgiedzinski) [inline src/modules/example/api/Controllers/ExampleController.cs:9]: I'm wondering why the code check doesn't work for grouping namespaces. Namespaces should be grouped and sorted.

File: src/api/app/Configuration/SwaggerConfigurationExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.OpenApi.Models;
+
+namespace Intive.Patronage2023.Api.Configuration;
+
+/// <summary>
+/// Static class SwaggerConfigurationExtensions.
```
Comment (mateuszgiedzinski) [inline src/api/app/Configuration/SwaggerConfigurationExtensions.cs:7]: I think better summary for that will be  Class that contains extension methods for swagger configuration.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/modules/example/api/Controllers/ExampleController.cs [hunk 5]
```diff
@@ -15,6 +16,7 @@ namespace Intive.Patronage2023.Modules.Example.Api.Controllers;
 /// Example controller.
 /// </summary>
 [ApiController]
+[Authorize]
```
Comment (mateuszgiedzinski) [inline src/modules/example/api/Controllers/ExampleController.cs:19]: Instead using Authorize everywhere we can specify filter for that. Read this article https://neelbhatt.com/2018/03/23/global-authorization-filter-in-net-core-net-core-security-part-v/
For now create simple policy that allow only authenticated users to use our api without any other requirements.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/api/app/Program.cs [hunk 6]
```diff
@@ -33,39 +36,33 @@
...
-	app.UseSwagger();
-	app.UseSwaggerUI();
-}
+	// turning off issuer validation and https
+	configureOptions.RequireHttpsMetadata = false;
+	configureOptions.TokenValidationParameters.ValidateIssuer = false;
+});
+builder.Services.AddAuthorization();
+
+builder.Services.AddMvc(options =>
```
Comment (mateuszgiedzinski) [inline src/api/app/Program.cs:57]: We don't need MVC here. We need only Controllers. You can add this lamda inside AddControllers

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

**Review by IzabelaPalubicka — APPROVED** [review]

**Review by mateuszgiedzinski — APPROVED** [review]

**Comment by mateuszgiedzinski** [comment]
Please resolve conflicts.

**Pull request merged**